app/services/ProbationOffenderSearch.java

```diff
@@ -0,0 +1,185 @@
+package services;
+
+import com.google.common.collect.ImmutableMap;
+import com.typesafe.config.Config;
+import controllers.OffenderController;
+import data.CourtDefendant;
+import data.MatchedOffenders;
+import helpers.Encryption;
+import helpers.JsonHelper;
+import helpers.JwtHelper;
+import interfaces.HealthCheckResult;
+import interfaces.OffenderApi;
+import interfaces.OffenderSearch;
+import interfaces.UserAwareApiToken;
+import lombok.val;
+import play.Logger;
+import play.libs.ws.WSClient;
+import play.libs.ws.WSResponse;
+import services.helpers.SearchQueryBuilder;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static interfaces.HealthCheckResult.healthy;
+import static interfaces.HealthCheckResult.unhealthy;
+import static play.mvc.Http.HeaderNames.AUTHORIZATION;
+import static play.mvc.Http.HeaderNames.CONTENT_TYPE;
+import static play.mvc.Http.MimeTypes.JSON;
+import static play.mvc.Http.Status.OK;
+
+public class ProbationOffenderSearch implements OffenderSearch {
+    private static final String CENTRAL_TEAM_CODE = "N40";
+    private final WSClient wsClient;
+    private final String apiBaseUrl;
+    private final UserAwareApiToken userAwareApiToken;
+    private final OffenderApi offenderApi;
+    private final Function<String, String> encrypter;
+
+
+    @Inject
+    public ProbationOffenderSearch(Config configuration, WSClient wsClient, UserAwareApiToken userAwareApiToken, OffenderApi offenderApi) {
+        apiBaseUrl = configuration.getString("probation.offender.search.url");
+        this.wsClient = wsClient;
+        this.userAwareApiToken = userAwareApiToken;
+        this.offenderApi = offenderApi;
+        encrypter = plainText -> Encryption.encrypt(plainText, configuration.getString("params.secret.key"))
+                .orElseThrow(() -> new RuntimeException("Encrypt failed"));
+    }
+
+    @Override
+    public CompletionStage<Map<String, Object>> search(String bearerToken, List<String> probationAreasFilter, String searchTerm, int pageSize, int pageNumber, SearchQueryBuilder.QUERY_TYPE queryType) {
+        return userAwareApiToken.get(JwtHelper.username(bearerToken))
```

Comment:
The response for the search API satisfies 90% of the Newtech requirements with these exceptions:

1. NewTech also requires links to the prisoner image
2. NewTech also requires descriptions for the probation area aggregations